models/fates/R/model2netcdf.FATES.R

```diff
@@ -10,581 +10,137 @@
+##' @param outdir Location of FATES model output (e.g. a path to a single ensemble output)
+##' @param sitelat Latitude of the site
+##' @param sitelon Longitude of the site
+##' @param start_date Start time of the simulation
+##' @param end_date End time of the simulation
+##' @param pfts a named vector of PFT numbers where the names are PFT names
+##'
 ##' 
 ##' @examples  
 ##' \dontrun{
 ##' example.output <- system.file("case.clm2.h0.2004-01-01-00000.nc",package="PEcAn.FATES")
 ##' model2netcdf.FATES(outdir="~/")
 ##' }
-##' 
-##' @export
 ##'
 ##' @author Michael Dietze, Shawn Serbin
-model2netcdf.FATES <- function(outdir) {
+## modified Yucong Hu 10/07/24
+##' 
+##' @export
 
-    # E.g. var_update("AR","AutoResp","kgC m-2 s-1", "Autotrophic Respiration")
-    # currently only works for xyt variables, need to expand to work for cohort-level outputs, 
-    # age bins, soils, etc
-    var_update <- function(out,oldname,newname,newunits=NULL,long_name=NULL){
-      if (oldname %in% ncin_names) {
-        ## define variable
-        oldunits <- ncdf4::ncatt_get(ncin,oldname,"units")$value
-        if (oldunits=="gC/m^2/s") oldunits <- "gC m-2 s-1"
-        if (oldname=="TLAI" && oldunits=="none") oldunits <- "m2 m-2"
-        if(is.null(newunits)) newunits = oldunits
-        newvar <- ncdf4::ncvar_def(name = newname, units = newunits, longname=long_name, dim = xyt)
-        
-        ## convert data
-        dat <- ncdf4::ncvar_get(ncin,oldname)
-        dat.new <- PEcAn.utils::misc.convert(dat,oldunits,newunits)
-        
-        ## prep for writing
-        if(is.null(out)) {
-          out <- list(var <- list(),dat <- list())
-          out$var[[1]] <- newvar
-          out$dat[[1]] <- dat.new
-        } else {
-          i <- length(out$var) + 1
-          out$var[[i]] <- newvar
-          out$dat[[i]] <- dat.new
-        }
+model2netcdf.FATES <- function(outdir, sitelat, sitelon, start_date, end_date, pfts) {
+  ## matched_var could be updated further to take in only oldnames users need.
+  matched_var <- list(c("FATES_GPP_PF","GPP","kgC m-2 s-1","Gross Primary Productivity"), 
+                      c("ER","TotalResp","kgC m-2 s-1","Total Respiration"),
+                      c("NEE","NEE","kgC m-2 s-1", "Net Ecosystem Exchange of carbon, includes fire and hrv_xsmrpool"),
+                      c("AR","AutoResp","kgC m-2 s-1","Autotrophic respiration (MR + GR)"),
+                      c("HR","HeteroResp","kgC m-2 s-1","Total heterotrophic respiration"),
+                      c("SR","SoilResp","kgC m-2 s-1","Total soil respiration (HR + root resp)"),
+                      c("TLAI","LAI","m2 m-2","Total projected leaf area index"),
+                      c("Qle","Evap","Evap","kgC m-2 s-1","Total evaporation"),
+                      c("QVEGT","Transp","kg m-2 s-1","Canopy transpiration")) 
+
+  var_update <- function(out,oldname,newname,newunits=NULL,long_name=NULL){
+    if (oldname %in% nc_month_names) {
```

Comment:
Note that `nc_month_names` and `nc_month` are not in the scope of this function (i.e. not passed as fcn args)